src/Components/TodoContainer.jsx

```diff
@@ -0,0 +1,195 @@
+import React from "react";
+import TodoList from "./TodoList";
+import AddTodoForm from "./AddTodoForm";
+import styles from "./TodoContainer.module.css";
+
+function TodoContainer() {
+  const [todoList, setTodoList] = React.useState([]);
+  const [isLoading, setIsLoading] = React.useState(true);
+  const [isAscending, setIsAscending] = React.useState(false);
+  const [isNewestFirst, setIsNewestFirst] = React.useState(false);
+
+  //Fetching data
+  async function fetchData() {
+    const url = `https://api.airtable.com/v0/${
+      import.meta.env.VITE_AIRTABLE_BASE_ID
+    }/${import.meta.env.VITE_TABLE_NAME}?view=Grid%20view`;
+
+    const options = {
+      method: "GET",
+      headers: {
+        Authorization: `Bearer ${import.meta.env.VITE_AIRTABLE_API_TOKEN}`,
+      },
+    };
+
+    try {
+      const response = await fetch(url, options);
+
+      if (!response.ok) {
+        const message = `Error: ${response.status}`;
+        throw new Error(message);
+      }
+
+      const data = await response.json();
+
+      const todos = data.records.map((todo) => {
+        return {
+          id: todo.id,
+          title: todo.fields.title,
+          createdTime: todo.createdTime,
+        };
+      });
+
+      setTodoList(todos);
+      setIsLoading(false);
+    } catch (error) {
+      console.log("Fetch error:", error.message);
+      return null;
+    }
+  }
+  //sort records
+  const handleSortByTitle = () => {
+    
+    const sortedTodos = [...todoList].sort((todoA, todoB) => {
+      const titleA = (todoA.title || "").toLowerCase();
+      const titleB = (todoB.title || "").toLowerCase();
+
+      if (titleA < titleB) return !isAscending ? -1 : 1;
+      if (titleA > titleB) return !isAscending ? 1 : -1;
+      return 0;
+    });
+
+    setTodoList(sortedTodos);
+    setIsAscending(!isAscending);
+  };
+
+  const handleSortByDate = () => {
+    const sortedTodos = [...todoList].sort((todoA, todoB) => {
+      const dateA = new Date(todoA.createdTime);
+      const dateB = new Date(todoB.createdTime);
+
+      return !isNewestFirst ? dateB - dateA : dateA - dateB;
+    });
+
+    setTodoList(sortedTodos);
+    setIsNewestFirst(!isNewestFirst);
+  };
+
+  // Fetch todos when the component mounts
+  React.useEffect(() => {
+    fetchData();
+  }, []);
+
+  //Add a new todo
+  async function addTodo(newTodoTitle) {
+    const url = `https://api.airtable.com/v0/${
+      import.meta.env.VITE_AIRTABLE_BASE_ID
+    }/${import.meta.env.VITE_TABLE_NAME}`;
+
+    const options = {
+      method: "POST",
+      headers: {
+        Authorization: `Bearer ${import.meta.env.VITE_AIRTABLE_API_TOKEN}`,
+        "Content-Type": "application/json",
+      },
+      body: JSON.stringify({
+        records: [
+          {
+            fields: {
+              title: newTodoTitle,
+              createdTime: new Date().toISOString(),
```

Comment:
Maybe my table in Airtable has a different structure, but with this line of code, when I try to add a new todo, I get a console error: 422 (The 422 status code indicates that the server was unable to process the request because it contains invalid data).
Airtable handles createdTime when you add a new todo to the database.

Reply:
My local and remote seemed not sync. Fixed it and removed date format from the code, could you try again?